src/PlaneWaveBasis.jl

```diff
@@ -255,15 +256,14 @@ function PlaneWaveBasis(model::Model{T}, Ecut::Number, fft_size, variational,
         Ecut, variational,
         opFFT, ipFFT, opBFFT, ipBFFT,
         r_to_G_normalization, G_to_r_normalization,
-        G_vectors(fft_size), r_vectors,
+        G_vects, r_vectors,
         kpoints, kweights_thisproc, kgrid, kshift,
         kcoords_global, kweights_global, comm_kpts, krange_thisproc, krange_allprocs,
         symmetries, symmetries_respect_rgrid, terms)
-
     # Instantiate the terms with the basis
     for (it, t) in enumerate(model.term_types)
         term_name = string(nameof(typeof(t)))
-        @timing "Instantiation $term_name" basis.terms[it] = t(basis)
+        @timing "Instantiation $term_name" basis.terms[it] = t(basis, array_type = array_type)
```

Comment:
this array type should be deduced from the basis array type

Reply:
Ok, but if I am not mistaken, we are going to loose some information about the G_vectors and r_vectors.
Doing 
```
G_vectors::AT
r_vectors::AT
```
will work, but not something like this
```
G_vectors::AT{Vec3{Int}, 3}
r_vectors::AT{Vec3{VT }, 3}
```

Reply:
Yes, that's true. In that case you would need two type arguments (one for `G_vectors::Garray` and one for `r_vectors::Rarray`) and then the `AT` could be implicitly obtained from the `G_vectors`.

Reply:
I'm not sure we want the two type arguments, however. @antoine-levitt thoughts?

Reply:
you can do {GT, RT, AT} where {GT <: AT, RT <: AT}. But yeah not great. OTOH we do want gvecs and rvecs to be statically typed, so yeah I don't think there's a way around it. What you can do is push the not interesting types further down the list, so you can do function f(basis::pwbasis{T, AT})